src/test/java/com/facebook/presto/hadoop/TestLineReader.java

```diff
@@ -0,0 +1,173 @@
+        InputStream in = new ByteArrayInputStream(input);
+        // Set the LineReader internal read buffer size 4 bytes
+        LineReader reader = new LineReader(in, 4);
+        Text str = new Text();
+        try {
+            reader.readLine(str, 30, 30);
+            // The LineReader does not store the new line character into str,
+            // and the str was doubled 3 times so its size is 32, with 5 0-valued
+            // bytes at the end, so we need to compare just the first 27 characters.
+            assertEquals(str.getLength(), 27);
+            assertEquals(Arrays.copyOfRange(str.getBytes(), 0, str.getLength()), Arrays.copyOfRange(input, 0, input.length - 1));
+        }
+        catch (IOException e) {
+            fail(e.getMessage());
+        }
+    }
+
+    @Test
+    public void testCustomReaderZeroBytes()
+    {
+        byte[] input = "Hello world! Goodbye world!\n".getBytes(UTF_8);
+        byte[] delimiter = "!".getBytes(UTF_8);
+        InputStream in = new ByteArrayInputStream(input);
+        // Set the LineReader internal read buffer size 4 bytes
+        LineReader reader = new LineReader(in, 4, delimiter);
+        Text str = new Text();
+        try {
+            // There should not be any bytes read into str because the maxLineLength is 0
+            reader.readLine(str, 0, 30);
+            assertEquals(str.getLength(), 0);
+            assertEquals(str.getBytes(), "".getBytes());
+        }
+        catch (IOException e) {
+            fail(e.getMessage());
+        }
+    }
+
+    @Test
+    public void testCustomReaderNonZeroBytes()
+    {
+        byte[] input = "Hello world! Goodbye world!\n".getBytes(UTF_8);
+        byte[] delimiter = "!".getBytes(UTF_8);
+        InputStream in = new ByteArrayInputStream(input);
+        // Set the LineReader internal read buffer size 4 bytes
+        LineReader reader = new LineReader(in, 4, delimiter);
+        Text str = new Text();
+        try {
+            // The first 11 bytes of input should be read into str because the 12th character is the delimiter
+            reader.readLine(str, 30, 30);
+            assertEquals(str.getLength(), 11);
+            assertEquals(Arrays.copyOfRange(str.getBytes(), 0, str.getLength()), "Hello world".getBytes(UTF_8));
+        }
+        catch (IOException e) {
+            fail(e.getMessage());
+        }
+    }
+
+    @Test
+    public void testDefaultReaderMaxBytesConsumed()
+    {
+        byte[] input = "Hello world! Goodbye world!\n".getBytes(UTF_8);
+        InputStream in = new ByteArrayInputStream(input);
+        // Set the LineReader internal read buffer size 4 bytes
+        LineReader reader = new LineReader(in, 4);
+        Text str = new Text();
+        try {
+            // The read should fail because the line length is greater than the maxBytesToConsume
+            reader.readLine(str, 0, 10);
+            fail("Expected exception " + IOException.class.getSimpleName());
```

Comment:
Sorry, I actually meant to just write
```java
fail("expected exception");
```
Alternatively, we could write
```java
fail("expected IOException");
```
While you often want to avoid hard-coding class names, in this case, there's no value for doing it given that "IOException" is part of the JDK and will never be renamed. Doing it that way here makes the code more complex and thus harder to read.